### ISSUE TITLE
Add new "elaboration" test type

### ISSUE THREAD

**Issue by tgorochowik** [chipsalliance/sv-tests#1770]
We do have an `can_elaborate` flag, but it's not enough. Elaboration should be a separate test type.

With current approach there are tests that will never pass for parsers, e.g see those added here: #1757

We should add a new "elaboration" test type and mark tests accordingly. Most of the tests can use both types (parsing and elaboration) so tools can choose how to run those tests.

**Comment by MikePopoloski** [comment]
I agree, and from the other direction the majority of the 'simulation' tests are actually just elaboration tests. Only a handful actually require simulation to pass.
